src/util/std_types.h

```diff
@@ -126,39 +110,36 @@ class symbol_typet:public typet
   }
 };
 
-/*! \brief Cast a generic typet to a \ref symbol_typet
- *
- * This is an unchecked conversion. \a type must be known to be \ref
- * symbol_typet.
- *
- * \param type Source type
- * \return Object of type \ref symbol_typet
- *
- * \ingroup gr_std_types
-*/
+/// \brief Cast a generic typet to a \ref symbol_typet.
+///
+/// This is an unchecked conversion. \param type must be known to be
+/// \ref symbol_typet.
+///
+/// \param type Source type.
+/// \return Object of type \ref symbol_typet.
 inline const symbol_typet &to_symbol_type(const typet &type)
 {
   PRECONDITION(type.id() == ID_symbol_type);
   return static_cast<const symbol_typet &>(type);
 }
 
-/*! \copydoc to_symbol_type(const typet &)
- * \ingroup gr_std_types
-*/
+/// \copydoc to_symbol_type(const typet &)
 inline symbol_typet &to_symbol_type(typet &type)
 {
   PRECONDITION(type.id() == ID_symbol_type);
   return static_cast<symbol_typet &>(type);
 }
 
+/// Check whether a reference to a generic typet is a \ref symbol_typet.
+/// \param type Source type.
+/// \return True if \param type is a \ref symbol_typet.
 template <>
 inline bool can_cast_type<symbol_typet>(const typet &type)
 {
   return type.id() == ID_symbol_type;
 }
 
-/*! \brief Base type of C structs and unions, and C++ classes
-*/
+/// Base type of C structs and unions, and C++ classes
```

Comment:
Java classes also derive from this, so if we're mentioning C++ classes we might as well include java here.